### DISCUSSION TITLE
How to make ticks display more reasonably in BarX

### DISCUSSION THREAD

**Discussion by Kiss-World** [observablehq/plot#2106] [Q&A] [answered]
y axis is too many
![image](https://github.com/user-attachments/assets/4f551875-577b-41ba-8273-6080711b54dd)

https://observablehq.com/d/0af4406113fad5e6

**Comment by that-ambuj** [comment]
Removing the `height` option in the plot would give it enough room to breath, but it'll take the whole page. 

You could also set y interval to some high number but some of your data points would be rounded, i guess.

Something like this:
```js
Plot.plot({
  y: { interval: 10 }
});
```

**Comment by Fil** [comment]
For a continuous *y* axis (instead of categorical), use rect + bin instead of bar + group:

```
   Plot.rectX(population, Plot.binY({x:'sum', interval: 1},{ …
```

**Comment by mbostock** [comment, marked as answer]
To illustrate the options described by @Fil and @that-ambuj, here is what rect + bin would look like:

![untitled - 2024-07-18T092910 017](https://github.com/user-attachments/assets/43ae489d-6891-4802-ac01-580f8e43d9fa)

```js
Plot.plot({
  height: 500,
  x: {
    label: "← men · population (thousands) · women →",
    labelAnchor: "center",
    tickFormat: Math.abs
  },
  marks: [
    Plot.rectX(
      population,
      Plot.binY(
        { x: "sum" },
        {
          x: (d) => d.population * (d.sex === "M" ? -1 : 1),
          y: "age",
          fill: "sex",
          interval: 1
        }
      )
    )
  ]
})
```

And here is what the **interval** option would look like (also using **reverse** to put zero at the bottom since ordinal *y* scales otherwise put the first value on top by default):

![untitled - 2024-07-18T093129 848](https://github.com/user-attachments/assets/443101de-e6ed-4bdd-a6d2-c6e99bdf4a2e)

```js
Plot.plot({
  height: 500,
  x: {
    label: "← men · population (thousands) · women →",
    labelAnchor: "center",
    tickFormat: Math.abs
  },
  y: {
    interval: 1,
    reverse: true
  },
  marks: [
    Plot.barX(
      population,
      Plot.groupY(
        { x: "sum" },
        {
          x: (d) => d.population * (d.sex === "M" ? -1 : 1),
          y: "age",
          fill: "sex"
        }
      )
    )
  ]
})
```

**Reply by Kiss-World** [reply]
Cool, thank you very much
